source/common/http/async_client_impl.cc

```diff
@@ -166,6 +169,8 @@ void AsyncStreamImpl::sendHeaders(RequestHeaderMap& headers, bool end_stream) {
   if (send_xff_) {
     Utility::appendXff(headers, *parent_.config_.local_info_.address());
   }
+
+  stream_info_.addBytesSent(headers.byteSize());
```

Comment:
Two options:
1. Rename this PR and update the commit to reflect the addition of stream info accounting. Add dedicated unit tests for just this.
2. Use existing stream_info dynamic metadata in options.metadata in your test and don't add in accounting stuff.